### PR TITLE
IGP 32

### PR DESCRIPTION
**Pull request by thrilok209**

### CODE REVIEW

File: contracts/payloads/IGP32/PayloadIGP32.sol [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+    /// @notice Action 2: Approve fUSDC and fUSDT protocols to spend the reserves tokens
+    function action2() internal {
+        address[] memory protocols = new address[](2);
+        address[] memory tokens = new address[](2);
+        uint256[] memory amounts = new uint256[](2);
+
+        // fUSDC
+        protocols[0] = F_USDC;
+        tokens[0] = USDC_ADDRESS;
+        amounts[0] = 250_000 * 1e6; // 250k USDC
```
Comment (bergben) [inline contracts/payloads/IGP32/PayloadIGP32.sol:520]: amounts here need to be 320k instead 

to cover currently left over rewards until 21 July + 250k new + buffer

File: contracts/payloads/IGP32/PayloadIGP32.sol [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+
+        // fUSDT
+        protocols[1] = F_USDT;
+        tokens[1] = USDT_ADDRESS;
+        amounts[1] = 250_000 * 1e6; // 250k USDT
+
+        FLUID_RESERVE.approve(protocols, tokens, amounts);
+    }
+
+    /// @notice Action 3: Dust reserves tokens allowance
```
Comment (bergben) [inline contracts/payloads/IGP32/PayloadIGP32.sol:530]: can we call this WBTC rewards reserves tokens allowance instead everywhere? 

just to properly differentiate between other approvals where we really only approve dust amounts

File: contracts/payloads/IGP32/PayloadIGP32.sol [hunk 3]
```diff
@@ -0,0 +1,563 @@
...
+    function action3() internal {
+        address[] memory protocols = new address[](2);
+        address[] memory tokens = new address[](2);
+        uint256[] memory amounts = new uint256[](2);
+
+        protocols[0] = 0x6F72895Cf6904489Bcd862c941c3D02a3eE4f03e;
+        tokens[0] = wBTC_ADDRESS;
+        amounts[0] = 0.8 * 1e8;
```
Comment (bergben) [inline contracts/payloads/IGP32/PayloadIGP32.sol:538]: maybe better to take 0.81 actually to give some buffer

File: contracts/payloads/IGP32/PayloadIGP32.sol [hunk 4]
```diff
@@ -0,0 +1,563 @@
...
+
+        protocols[1] = 0x3A0b7c8840D74D39552EF53F586dD8c3d1234C40;
+        tokens[1] = wBTC_ADDRESS;
+        amounts[1] = 0.8 * 1e8;
+
```
Comment (bergben) [inline contracts/payloads/IGP32/PayloadIGP32.sol:543]: can we also add borrow token dust amounts?
```suggestion
        protocols[2] = 0x6F72895Cf6904489Bcd862c941c3D02a3eE4f03e;
        tokens[2] = USDC_ADDRESS;
        amounts[2] = 100 * 1e6;
                
        protocols[3] = 0x3A0b7c8840D74D39552EF53F586dD8c3d1234C40;
        tokens[3] = USDT_ADDRESS;
        amounts[3] = 100 * 1e6;
```

borrow token should not be needed to be transferred in when doing rebalance() but just to be safe and guarantee filling WBTC rewards will work as we anyway usually give dust allowance

File: contracts/payloads/IGP32/PayloadIGP32.sol [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+            description
+        );
+
+        require(proposedId == PROPOSAL_ID, "PROPOSAL_IS_NOT_SAME");
+    }
+
+    function execute() external {
+        require(address(this) == address(TIMELOCK), "not-valid-caller");
+
+        // Action 1: Deploy wBTC/ETH and ETH/wBTC vaults.
```
Comment (KaymasJain) [inline contracts/payloads/IGP32/PayloadIGP32.sol:484]:
```suggestion
        // Action 1: Update rewards for fUSDC & fUSDT.
```

File: contracts/payloads/IGP32/PayloadIGP32.sol [hunk 6]
```diff
@@ -0,0 +1,563 @@
...
+        require(address(this) == address(TIMELOCK), "not-valid-caller");
+
+        // Action 1: Update rewards for fUSDC & fUSDT.
+        action1();
+
+        // Action 2: Approve fUSDC and fUSDT protocols to spend the reserves tokens
+        action2();
+
+        // Action 3: Dust reserves tokens allowance
+        action2();
```
Comment (KaymasJain) [inline contracts/payloads/IGP32/PayloadIGP32.sol:491]:
```suggestion
        action3();
```
Also, need to add action4();

File: contracts/payloads/IGP32/PayloadIGP32.sol [hunk 7]
```diff
@@ -0,0 +1,563 @@
...
+    function execute() external {
+        require(address(this) == address(TIMELOCK), "not-valid-caller");
+
+        // Action 1: Update rewards for fUSDC & fUSDT.
+        action1();
+
+        // Action 2: Approve fUSDC and fUSDT protocols to spend the reserves tokens
+        action2();
+
+        // Action 3: Dust reserves tokens allowance
```
Comment (KaymasJain) [inline contracts/payloads/IGP32/PayloadIGP32.sol:490]:
```suggestion
        // Action 3: WBTC reserves tokens allowance to WBTC/USDC & WBTC/USDT vault
```

**Pull request merged**